beam-sync.md

```diff
@@ -0,0 +1,187 @@
+# Beam Sync
+
+## Intro
+
+Beam Sync is a new strategy to transition from an empty client node to a full-state, fully-validating
+node on the network. It borrows heavily from the current "Fast Sync" approach.
+
+#### Fast Sync
+
+This writeup assumes familiarity with Fast Sync and supporting devp2p protocols like `GetNodeData`.
+
+#### Beam Sync Benefits
+
+Beam Sync provides several important benefits, to both the syncing node, and the broader network health.
+For example:
+
+- The "Fast" syncing experience is very slow, taking days before any queries
+  or validation can be done on the new node. Beam Sync has the promise of syncing to a useable node in minutes.
+- Fast Sync nodes are pure leechers on the network. Beam Sync nodes are very effective at seeding
+  data to other Beam-syncing nodes.
+- Fast Sync gets more likely to fault as total state size grows. Beam Sync gets more likely to fault
+  as state reads per block grows.
+
+#### Beam Sync Approach
+
+Roughly, the idea is for a client to skip to the head of the chain (like Fast Sync). Unlike
+Fast Sync, the block starts executing immediately, collecting *only* the state data needed
+to execute & verify the block. Crucially, a background process collects data that is not
+being read by any block. When this background process is complete, then all of Beam Sync is
+complete, and the node transitions to Full Sync.
+
+## How to Beam Sync
+
+There are many possible approaches to Beam Sync. The plan is to evolve Beam Sync through three stages:
+
+- v0: Use existing Fast Sync protocols; works on mainnet today
```

Comment:
Maybe instead of `v0` we should use the terminology `phase 0`?  They feel less like versions and more like "phases" but this is absolutely semantics nitpickery.